### PR TITLE
Testing CentOS7 clang 3.5 without fortify

### PR DESCRIPTION
**Pull request by theopolis**
A working test of CentOS7 without fortify, which causes NL socket reads to hang.

### CODE REVIEW

**Comment by javuto** [comment]
ok to test

**Comment by theopolis** [comment]
@javuto, I'm testing installing boost/thrift/rocksdb (not using the yum-provided versions) just for fun. I've tried with 3.4, 3.5, 3.6 versions of LLVM/clang and no dice.

File: CMakeLists.txt [hunk 1]
```diff
@@ -28,26 +33,42 @@ if(APPLE)
...
+  set(C_COMPILE_FLAGS "${C_COMPILE_FLAGS} -g -DDEBUG -O0 -pg")
+elseif(DEFINED ENV{SANITIZE})
+  # make santifize (cannot make debug sanitize)
+  set(C_COMPILE_FLAGS "-g -O0 -fno-omit-frame-pointer")
+  set(C_COMPILE_FLAGS "${C_COMPILE_FLAGS} -fsanitize=leak -fsanitize=address")
 else()
-  set(C_COMPILE_FLAGS "${CMAKE_C_FLAGS} -O2")
+  set(C_COMPILE_FLAGS "${C_COMPILE_FLAGS} -O2")
+  # Do not enable fortify with clang: http://llvm.org/bugs/show_bug.cgi?id=16821
+  #set(C_COMPILE_FLAGS "${C_COMPILE_FLAGS} -D_FORTIFY_SOURCE=2")
```
Comment (theopolis) [inline CMakeLists.txt]: Unfortunate, but on CentOS7 this is optimizing out parts of code and causing infinite jumps.

**Comment by theopolis** [comment]
Extensions test has confounds, looks good, @javuto can we make clean on the CentOS7 to get rid of all of those CMake warnings from the CMakeCache?

**Pull request merged**